internal/services/serviceconnector/service_connector_app_service_resource.go

```diff
@@ -89,6 +92,62 @@ func (r AppServiceConnectorResource) Arguments() map[string]*schema.Schema {
 		},
 
 		"authentication": authInfoSchema(),
+
+		"scope": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"configuration": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"action": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(servicelinker.ActionTypeOptOut),
+							string(servicelinker.ActionTypeEnable),
+						}, false),
+					},
+
+					"configuration_store": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"app_configuration_id": {
+									Type:     pluginsdk.TypeString,
+									Optional: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"public_network_solution": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"action": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(servicelinker.ActionTypeOptOut),
+							string(servicelinker.ActionTypeEnable),
+						}, false),
+					},
+				},
+			},
+		},
```

Comment:
I missed this on the last review, but are we expecting this block to have further properties in future? Having a block/list, with `MaxItems = 1` and only 1 property feels redundant, and could be flattened to a simple property? WDYT?

Reply:
Right, this block do have some other properties will be supported in the future.